### ISSUE TITLE
a11y check

### ISSUE THREAD

**Issue by asangma** [Esri/global-nav#30]
- [ ] Footer
- [ ] Header
- [ ] Language Switcher

cc @jonathantneal @HeathMeyette

**Comment by jimmieego** [comment]
**Header**

The user thumbnail does not have `alt`, although this does not cause VoiceOver to announce the image file name. 

![screen shot 2017-08-01 at 1 37 56 pm](https://user-images.githubusercontent.com/5437651/28846059-e30865ae-76be-11e7-8735-261daed1041a.png)

**Comment by jonathantneal** [comment]
@jimmieego, thanks for looking at the header!

To address your comment; the `<img>` should not have an `alt`, or so I think, as it is presentational within a `<button>`, and the button is announced as `Account Profile`.

Was there an AT announcing the contents of the button? Would you have other concerns?

**Comment by jimmieego** [comment]
@jonathantneal I think we just need set the `alt=""` if it is presentational, right? VoiceOver announces the button as "Account Profile, collapsed, popup button" which I think is the correct behavior.

**Comment by jimmieego** [comment]
**Footer**
Starting from the top of the  test page (https://esri.github.io/global-nav/), I cannot tab to the footer. If I use mouse to scroll down and let the footer show up in the viewport, I can then tab through the links in the footer. This may be because the footer is initially hidden.

**Comment by jimmieego** [comment]
**Language Switcher**

The language switch dialog does not trap focus within the dialog.

**Comment by jimmieego** [comment]
**Search**

The "Search Esri.com" `label` has `for = "esri-header-search-dialog-query-control"`, but it looks like there isn't an element in the DOM with `id ="esri-header-search-dialog-query-control"`? The `for` should be probably set to the search input's id, which is `esri-header-search-dialog-control`.

**Comment by jonathantneal** [comment]
> I think we just need to set the alt="" if it is presentational, right

In your tests, was it announced? It should already be presentational by ancestry.

> I cannot tab to the footer

Great catch. While most pages will have something to tab to that adjusts the scroll before the global footer is reached, I will switch to using opacity for the global footer so that it is still accessible via keyboard navigation.

> The language switch dialog does not trap focus within the dialog.

That’ll be fun to fix. I know a few techniques. None are *perfect*, though in a year or two the `inert` attribute will make this trivial.

> it looks like there isn't an element in the DOM [with the right id]

Great catch, again. I’ll fix that. Thanks!

**Comment by asangma** [comment]
@jimmieego Feel free to close when the a11y issues are handled. :)

**Comment by jonathantneal** [comment]
@jimmieego, thanks for the feedback! And for the testing!

I leverage the `button` role and `aria-label` (of `Account Profile`) to announce just that, and skip over a more detailed reading of the button — this information is available upon expanding the menu.

I was concerned that method was not enough until you mentioned that it was development tooling that flagged it. I have had similar issues with tooling (and I love tooling, and I have absolutely nothing against tooling). Will you also be testing with JAWS, VoiceOver, NVDA (, ChromeVox, Windows Eyes 9, etc.)?

**Comment by jimmieego** [comment]
@jonathantneal I've tested it with VoiceOver, NVDA, and ChromeVox. Unfortunately I don't have access to JAWS. I am not familiar with Windows Eyes 9. What is that?

**Comment by jimmieego** [comment]
FYI, a11y cast has a video about modal dialogs: https://youtu.be/JS68faEUduk
